BaragonServiceIntegrationTests/src/test/java/com/hubspot/baragon/BaragonServiceTestIT.java

```diff
@@ -0,0 +1,478 @@
+  
+  private void setupTestService(String requestId, String serviceId, String basePath, Optional<String> replaceServiceId, Optional<Map<String, Object>> options, Optional<String> template) {
+    System.out.println("Requesting " + requestId + "...");
+    UpstreamInfo upstream = new UpstreamInfo(UPSTREAM, Optional.<String>absent(), Optional.<String>absent());
+    BaragonService lbService = new BaragonService(serviceId, new ArrayList<String>(), basePath, 
+        new HashSet<String>(Arrays.asList(LOAD_BALANCER_GROUP)), options.isPresent() ? options.get() : new HashMap<String, Object>(), template);
+    
+    BaragonRequest request = new BaragonRequest(requestId, lbService, Arrays.asList(upstream), new ArrayList<UpstreamInfo>(), replaceServiceId);
+    baragonServiceClient.enqueueRequest(request);
+  }
+  
+  private void removeService(String serviceId) {
+    System.out.println("Cleaning up...");
+    BaragonResponse resp = baragonServiceClient.deleteService(serviceId).get();
+    while(baragonServiceClient.getRequest(resp.getLoadBalancerRequestId()).get().getLoadBalancerState().equals(BaragonRequestState.WAITING)) { }
+    
+    assertFalse(baragonServiceClient.getServiceState(serviceId).isPresent());
+  }
+  
+  // ----------------------------------------------------
+  
+  @Before
+  public void setup() throws Exception {
+    baragonServiceClient = getInjector().getInstance(BaragonServiceClient.class);
+  }
+  
+  @Rule
+  public TestRule watcher = new TestWatcher() {
+     protected void starting(Description description) {
+        System.out.println("Starting: " + description.getMethodName());
+     }
+     protected void succeeded(Description description) {
+       System.out.println("\u001B[32mTest passed\u001B[0m");
+     }
+     protected void failed(Throwable e, Description description) {
+       System.out.println("\u001B[31mTest failed\u001B[0m");
+     }
+  };
+  
+  
+  @Rule
+  public Stopwatch stopwatch = new Stopwatch() {
+      @Override
+      protected void finished(long nanos, Description description) {
+        System.out.println("(" + nanos / 1000000000.0 + " s)");
+      }
+  };
+  
+  @After
+  public void teardown() {
+    baragonServiceClient = null;
+  }
+
+  // ----------------------------------------------------
+  
+  @Test
+  public void testStatus() throws Exception {
+    Optional<BaragonServiceStatus> status = baragonServiceClient.getAnyBaragonServiceStatus();
+    assertTrue(status.get().isLeader());
+  }
+
+  @Test
+  public void testWorkers() throws Exception {
+    assertFalse(baragonServiceClient.getBaragonServiceWorkers().isEmpty());
+  }
+  
+  @Test
+  public void testLoadBalancers() throws Exception {
+    assertFalse(baragonServiceClient.getLoadBalancerGroups().isEmpty());
+  }
+  
+  @Test
+  public void testClusterAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
```

Comment:
Rather than picking one to test, why don't we iterate over all of the available agents here? We are only testing one/two now, but if we add more, they should definitely all have the proper info